docs/examples/nodejs-read-index-layer.md

```diff
@@ -0,0 +1,236 @@
+     customUrl: "http://YourCustomEnvironment",
+     credentials: credentials,
+     tokenRequester: requestToken
+   });
+   ```
+
+5. Get the OAuth 2.0 token from OLP using the `getToken` method.
+
+   ```typescript
+   const token: string = await userAuth.getToken();
+   ```
+
+You can use the `UserAuth` instance to create the `OlpClientSettings` object.
+
+## <a name="create-olpclientsettings"></a>Create `OlpClientSettings`
+
+You need to create the `OlpClientSettings` object to get catalog and partition metadata, as well as layer data from the Open Location Platform (OLP).
+
+To create the `OlpClientSettings` object:
+
+1. [Authenticate](#authenticate-using-client-credentials) to the Open Location Platform (OLP).
+
+2. Import the `OlpClientSettings` class from the `olp-sdk-dataservice-read` module.
+
+   ```typescript
+   import { OlpClientSettings } from "@here/olp-sdk-dataservice-read";
+   ```
+
+3. Create the `olpClientSettings` instance using the environment in which you work and the `getToken` method.
+
+   ```typescript
+   const olpClientSettings = new OlpClientSettings({
+     environment:
+       "here | here-dev | here-cn | here-cn-dev | http://YourCustomEnvironment",
+     getToken: () => userAuth.getToken()
+   });
+   ```
+
+## <a name="create-IndexlayerClient"></a>Create IndexLayerClient`
+
+You can use the `IndexLayerClient` object to request any data and partition metadata from a [index layer](https://developer.here.com/olp/documentation/data-user-guide/portal/layers/creating-index-layer.html).
+
+To create the `IndexLayerClient` object:
+
+1. Get an access key ID and access key secret.
+   For instructions, see [Authenticate to HERE OLP Using Client Credentials](#authenticate-using-client-credentials).
+
+2. Create the `OlpClientSettings` object.
+   For instructions, see [Create OlpClientSettings](#create-olpclientsettings).
+
+3. Create the `IndexLayerClient` object with the HERE Resource Name (HRN) of the catalog that contains the layer, the layer ID, and the OLP client settings from step 2.
+
+   ```typescript
+   const indexLayerClient = await new IndexLayerClient()
+       hrn: "CatalogHRN",
+       layerId: "LayerId",
+       olpClientSettings
+   );
+   ```
+
+## Get Partition Metadata from an Index Layer
+
+Partition metadata from an index layer consists of the following information about the partition:
+
+- ID (data handle)
+- Data size
+- Checksum
+- Metadata
+- Timestamp
+
+To get partition metadata from an index layer:
+
+1. Create the `IndexLayerClient` object.
+   For instructions, see [Create IndexLayerClient](#create-IndexlayerClient).
+
+2. Create the `IndexQueryRequest` object with the `RSQL query` to retrieve partitions that match the query and `huge` boolean parameter setting `true` if query is huge. More information could be find [Get Data from an Index Layer](https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/getting-data-index.html).
```

Comment:
```suggestion
2. Create the `IndexQueryRequest` object with the RSQL query string and, if the query string is huge, set the `huge` boolean parameter to `true` .
```
Why do we need a reference to the similar instruction? Is there anything in the referenced instruction that should be mentioned here?

Reply:
Fixed